src/data.rs

```diff
@@ -91,7 +94,8 @@ impl DataType {
         self.file_handle = handle;
     }
 
-    pub fn init_data_type(&mut self, param: InitParams) -> Result<()> {
+    pub fn init_data_type(&mut self, param: InitParams) -> Result<u128> {
+        let start = Instant::now();
```

Comment:
I don't think these functions should be measuring and returning their own time. It should be measured by the caller.